src/Illuminate/Http/Resources/Json/JsonResource.php

```diff
@@ -218,7 +226,34 @@ public function response($request = null)
      */
     public function toResponse($request)
     {
-        return (new ResourceResponse($this))->toResponse($request);
+        return $this->resolveResourceResponse()
+            ->toResponse($request);
+    }
+
+    /**
+     * @return ResourceResponse
+     */
+    protected function resolveResourceResponse()
+    {
+        if ($this->resourceResponse) {
+            $resourceResponse = $this->resourceResponse;
+        } else {
+            $resourceResponse = $this->isResourcePaginated()
+                ? PaginatedResourceResponse::class
+                : ResourceResponse::class;
```

Comment:
Why did you change the behavior of the base JsonResource class to do this pagination stuff? It was not like that before. Please *only* make the changes necessary for the feature you want.